src/toxcore/tcp/server/server.rs

```diff
@@ -1237,4 +1267,30 @@ mod tests {
             .wait();
         assert!(handle_res.is_err());
     }
+    #[test]
+    fn tcp_ping_sender_test() {
+        let server = Server::new();
+
+        // client #1
+        let (mut client_1, rx_1) = create_random_client();
+        client_1.set_last_pinged(Instant::now().sub(Duration::from_secs(TCP_PING_FREQUENCY + 1)));
```

Comment:
It won't always work - this sub can cause an integer overflow. See tokio_timer::clock, for example: https://github.com/tox-rs/tox/blob/3137d7a32d43a7a3971a6cda9dc0d40ef328cc11/src/toxcore/onion/onion_announce.rs#L565-L581